crates/core_arch/src/macros.rs

```diff
@@ -9,34 +9,98 @@ impl<const imm: i32, const bits: i32> ValidateConstImm<imm, bits> {
     };
 }
 
-#[allow(unused)]
+#[allow(non_camel_case_types)]
+pub(crate) struct ValidateConstImm_S<const imm_s5: i32, const min: i32, const max: i32>();
+impl<const imm: i32, const min: i32, const max: i32> ValidateConstImm_S<imm, min, max> {
+    pub(crate) const VALID: () = {
+        let _ = 1 / ((imm >= min && imm <= max) as usize);
+    };
+}
```

Comment:
I would prefer modifying the main `ValidateConstImm` to support both a min and max value.